### PR TITLE
small improvements

### PR DESCRIPTION
**Pull request by MCWertGaming**
Hello,
at playing a bit around with you library, i noticed some bad practices and old standard I wan't to improve with this pull-request. Cpp-terminal is using some really old and deprecated headers, I changed them to use the new (C++ exclusive) ones. There are some other things too, like using for-loops instead of range-based loops, making class functions const instead of static and unsing null instead of the nullptr. 
Feel free to request changes or add comments! I'm lucky to help.

Cheers!

### CODE REVIEW

**Comment by certik** [comment]
Hi Damon, thank you! All looks good. There is a failure on Windows, but I can't see from my phone what it is --- is it caused by this PR?

**Comment by MCWertGaming** [comment]
Yes, i'm sorry for that. I fixed it already! Waiting for the CI to finish.

**Comment by MCWertGaming** [comment]
Okay, it works now! I quickly want to note that i removed the _WINSOCKAPI_ definition, because it's not used anywhere. If it's important for something, I will add it again.

File: cpp-terminal/terminal.h
Comment (certik) [inline cpp-terminal/terminal.h:353]: If you remove the `return`, I think it's a good habit to document that the `case` is meant to fall through:
```suggestion
            case '\x0a': // LF; falls-through
```

File: cpp-terminal/terminal.h
Comment (certik) [inline cpp-terminal/terminal.h:408]: The way `cpp-terminal` is designed is that the `terminal.h` file is strictly platform independent, and thus does not have any kind of platform specific `ifdef` statements. The file `terminal_base.h` is platform dependent. As such, this should be moved there in some way. Do you think you could please do that?

Why is the `sscanf_s` needed? I thought Windows tests are passing in master with the `sscanf` function.

File: examples/kilo.cpp
Comment (certik) [inline examples/kilo.cpp:5]: The file `kilo.cpp` is taken from elsewhere, I put it here to show that one can take a linux specific terminal application and easily transform it to use `cpp-terminal` to make it multiplatform. Most of the changes below are for the upstream code at https://github.com/snaptoken/kilo-src/blob/bc1039e7e2ac49ca6d0628ccd76c699b0736e1f4/kilo.c. I fine with the changes, they are relatively small.

**Review by certik — CHANGES_REQUESTED** [review]
Thank you @MCWertGaming for submitting this PR, all the changes look good. I left some minor comments. There is only one change that should be done before we can merge it, moving the `ifdef` from `terminal.h` to `terminal_base.h`, see the above comment for more details.

Let me know if you need any help with this. Thank you again!

File: cpp-terminal/terminal.h
Comment (MCWertGaming) [reply cpp-terminal/terminal.h:408]: Yes, i will try to move that. The thing with sscanf is, that MSVC compiles it just fine, but throws a warning. That's, because microsoft provides their own implementation (sscanf_s) which is probably safer in terms of conversion errors. Linux don't offers that function, but clang-tidy notes:
```
'sscanf' used to convert a string to an integer value, but function will not report conversion errors; consider using 'strtol' instead
```
So i think, it would be a great thing to implement sscanf_s on windows and strtol on linux (even though strtol will require more effort because of different arguments).

File: examples/kilo.cpp
Comment (MCWertGaming) [reply examples/kilo.cpp:5]: Nice to know that. I would have offered to try on merging it into upstream, but the new headers are part of std and therefore C++ exclusive.

File: cpp-terminal/terminal.h
Comment (certik) [reply cpp-terminal/terminal.h:408]: What is the C++ way to do what I am doing there? Using the regex standard library?

Comment (MCWertGaming) [reply cpp-terminal/terminal.h:408]: I'm not sure, if there is a "C++ way of string conversion", the c++ reference [(here)](http://www.cplusplus.com/reference/cstdio/sscanf/) lists it inside both <stdio.h> and <cstdio>, so it's actually inside the C++ standard. Ofcourse you could say, that using the stdio library is bad, because it just offers some C functions inside C++ and is in fact not a "real" part of the C++ standard namespace. I have thought about maybe using std::stringstream. I will investigate both solutions and review my findings later.
Thank you.

Comment (certik) [reply cpp-terminal/terminal.h:408]: The easiest to get this PR merged is to simply provide our own `sscanf` method in `terminal_base.h` that dispatches to `sscanf_s` on Windows using `ifdef`. Then we simply call it from `terminal.h`.

Later, we can figure out if there is a better way.

**Comment by MCWertGaming** [comment]
Everything should be fine now! You can merge it, if everything is as it should be ^^

Damon

**Comment by certik** [comment]
Thanks for the fixes, everything looks good.

I see you introduced #pragma once. According to

https://stackoverflow.com/questions/787533/is-pragma-once-a-safe-include-guard

That is not standard C++. So I would recommend to sticking to the standardized subset of C++. What do you think?

**Review by certik — APPROVED** [review]
Everything looks good now, thank you for fixing it!

**Pull request merged**
